ci_framework/roles/dlrn_promote/tasks/promote_hash.yml

```diff
@@ -0,0 +1,31 @@
+---
+- name: Promote target hash
+  ansible.builtin.shell:
+    cmd: >-
+      export SSL_CA_BUNDLE={{ cifmw_dlrn_promote_ssl_ca_bundle }};
+      dlrnapi --url {{ cifmw_repo_setup_dlrn_api_url }}
+      {% if cifmw_dlrn_report_dlrnapi_host_principal is defined and cifmw_dlrn_report_kerberos_auth|bool -%}
+      --server-principal {{ cifmw_dlrn_report_dlrnapi_host_principal }} --auth-method kerberosAuth
+      {% endif -%}
+      repo-promote
+      {% if (cifmw_repo_setup_dlrn_hash_tag is defined) and (cifmw_repo_setup_dlrn_hash_tag | length > 0) -%}
+      --agg-hash {{ cifmw_repo_setup_dlrn_hash_tag }}
+      {% elif (cifmw_repo_setup_full_hash is defined) and (cifmw_repo_setup_full_hash | length > 0) -%}
+      --agg-hash {{ cifmw_repo_setup_full_hash }}
+      {% endif -%}
+      {% if (cifmw_repo_setup_extended_hash is defined) and (cifmw_repo_setup_extended_hash | length > 0) -%}
+      --extended_hash {{ cifmw_repo_setup_extended_hash }}
+      {% endif -%}
+      {% if (cifmw_repo_setup_commit_hash is defined) and (cifmw_repo_setup_commit_hash | length > 0) -%}
+      --commit_hash {{ cifmw_repo_setup_commit_hash }}
+      {% endif -%}
+      {% if (cifmw_repo_setup_distro_hash is defined) and (cifmw_repo_setup_distro_hash | length > 0) -%}
+      --distro_hash {{ cifmw_repo_setup_distro_hash }}
+      {% endif -%}
+      --promote-name { cifmw_dlrn_promote_promotion_target }}
+  environment: |
```

Comment:
Same comments as for the vars dict, we put vars and envs always on top. I'd try to be consistent across the ci-framework.

Reply:
Done